splink/diagnostics.py

```diff
@@ -0,0 +1,158 @@
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.session import SparkSession
+from pyspark.context import SparkContext, SparkConf
+from pyspark.sql import SQLContext
+from pyspark.sql.types import StructType, StructField, StringType, DoubleType, Row
+import pyspark.sql.functions as f
+from pyspark.sql.functions import when
+from .check_types import check_types
+
+
+import pandas as pd
+import warnings
+
+
+altair_installed = True
+try:
+    import altair as alt
+except ImportError:
+    altair_installed = False
+
+
+hist_def_dict = {
+    "$schema": "https://vega.github.io/schema/vega-lite/v4.8.1.json",
+    "config": {
+        "title": {"fontSize": 14},
+        "view": {"continuousHeight": 300, "continuousWidth": 400},
+    },
+    "data": {"values": None},
+    "encoding": {
+        "tooltip": [{"field": "count_rows", "title": "count", "type": "quantitative"}],
+        "x": {
+            "axis": {"title": "splink score"},
+            "bin": "binned",
+            "field": "splink_score_bin_low",
+            "type": "quantitative",
+        },
+        "x2": {"field": "splink_score_bin_high"},
+        "y": {
+            "field": "normalised",
+            "type": "quantitative",
+            "axis": {"title": "Probability density"},
+        },
+        "height": 200,
+        "mark": "bar",
```

Comment:
I don't think this schema is correct - I think height, mark, title and width go outside the 'encoding' key

Reply:
```
hist_def_dict = {
    "$schema": "https://vega.github.io/schema/vega-lite/v4.8.1.json",
    "config": {
        "title": {"fontSize": 14},
        "view": {"continuousHeight": 300, "continuousWidth": 400},
    },
    "data": {"values": None},
    "height": 200,
        "mark": "bar",
        "title": "Estimated Probability Density",
        "width": 700,
    "encoding": {
        "tooltip": [{"field": "count_rows", "title": "count", "type": "quantitative"}],
        "x": {
            "axis": {"title": "splink score"},
            "bin": "binned",
            "field": "splink_score_bin_low",
            "type": "quantitative",
        },
        "x2": {"field": "splink_score_bin_high"},
        "y": {
            "field": "normalised",
            "type": "quantitative",
            "axis": {"title": "Probability density"},
        },
        
    },
}
```